### ISSUE TITLE
Better Error Handling for Scan Upload

### ISSUE THREAD

**Issue by MichaelGissingNC** [DependencyTrack/dependency-track#162]
I just tried to manually upload a dependency check XML. On the server I just get the following messages:

```
2018-06-07 13:54:53,435 [] ERROR [org.owasp.dependencytrack.tasks.ScanUploadProcessingTask] Error while processing scan result
2018-06-07 13:54:53,435 [] ERROR [org.owasp.dependencytrack.tasks.ScanUploadProcessingTask] null
```

The corresponding code is:

```
} catch (Exception ex) {
    LOGGER.error("Error while processing scan result");
    LOGGER.error(ex.getMessage());
}
```

This is not 100% useful. For such cases I prefer to log the whole exception, potentially in a way that the full stack trace is only printed to the log if debug is enabled. But with this code there is no chance to get any deeper insight.

----

Just for completeness: haven't found the root cause yet, here's my script, if someone wants to look at it:

```
cat > payload.json <<__HERE__
{
  "project": "...",
  "scan": "$(cat depcheck/dependency-check-report.xml |base64 -w 0 -)"
}
__HERE__

curl -X "PUT" "https://..../dependency-track/api/v1/scan" \
     -k \
     -H 'Content-Type: application/json' \
     -H 'X-API-Key: ...' \
     -d @payload.json

```

**Comment by iabudiab** [comment]
@MichaelGissingNC Seems to be related to/duplicate of #156, #157 and #159 

You could try the 3.1 snapshot for the time being. I'm also waiting for the 3.1 release in order to integrate DTrack into our CI. Currently running the snapshot on staging.

**Comment by MichaelGissingNC** [comment]
@iabudiab thanks for the comment. I will continue investigating. However, my intention for this issue/report is not to discuss the actual root cause, but to state my desire for more useful error message :)

**Comment by iabudiab** [comment]
@MichaelGissingNC makes sense, a more useful debug log would help enormously . Just wanted to spare you the time in your investigations :wink:

**Comment by MichaelGissingNC** [comment]
Thanks Steve, helps a lot.

**Comment by stevespringett** [comment]
Included in v3.1.0. Closing.
